app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java

```diff
@@ -265,15 +265,20 @@ public void notificationSetter(Context context) {
         }
     }
 
-    private void  ArmTimer(Context context, int time) {
+    private void  ArmTimer() {
         Log.e(TAG, "ArmTimer called");
-        if(ActiveBgAlert.getOnly() != null) {
-            Calendar calendar = Calendar.getInstance();
-            AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
-            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
-                alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
-            } else {
-                alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+        ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
+        if(activeBgAlert != null ) {
+            AlertType alert = AlertType.get_alert(activeBgAlert.alert_uuid);
+            if(alert != null) {
+                int time = alert.minutes_between;
+                Calendar calendar = Calendar.getInstance();
+                AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+                    alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+                } else {
+                    alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+                }
             }
```

Comment:
Main issue that I see here is that there is going to be an alert every time there is a reading. So, if you put this on 20 minutes, you will still get the alert every 5 minutes. So maybe we should limit minutes_between to 5 minutes. even in that case, there is a chance that the alerts and the time of the reading will make it like there are two alerts every 5 minutes.
So maybe we should live the timer as it is today on one minute, but before playing anything check that we haven't played anything on the minutes_between?